### DISCUSSION TITLE
Adding shaders from MathBox

### DISCUSSION THREAD

**Discussion by kskyten** [JuliaPlots/Makie.jl#4173] [Ideas]
[MathBox](https://gitgud.io/unconed/mathbox) is a 3D math graphing library that can produce really impressive visualizations. The library was designed to make mathematical presentations. It has its own system for wiring shaders together and handling transitions between slides. The design of the library is discussed in [here](http://acko.net/blog/mathbox2/).

I know very little about graphical programming, but would like to add MathBox's functionality to Makie.  As far as I understand, copying the [shader code](https://gitgud.io/unconed/mathbox/tree/master/src/shaders/glsl) should be straightforward. Wiring the shaders together and abstracting this into a nice interface is another matter. How do I get started?

**Comment by SimonDanisch** [comment]
You could start by posting the coolest applications and shaders you want to have ;) Best with pictures !

**Comment by kskyten** [comment]
This slides 40-59 of [this](http://acko.net/files/gltalks/pixelfactory/online.html#40) slideshow is the kind of thing I'd like to be able to do. I'd like to use Julia packages directly instead of having to translate them to javascript first. I'll have to reverse engineer the presentation to figure out which shaders I need. The transition handling is also really cool (switch between slides fast to see this).

**Comment by kskyten** [comment]
I think I'll try to use one of the javascript packages in Julia as a front end for MathBox first. I'll see how far I can get by just pushing the data to be plotted from Julia.

**Comment by SimonDanisch** [comment]
Probably sensible for now... If you do something cool that you'd like in Makie, post it here ;) We can try to figure out what would be needed for it!
